internal/test/test.go

```diff
@@ -119,84 +140,110 @@ func run(
 		}
 	}
 
-	return &result{
-		Results:        res,
-		CoverageReport: coverageReport,
-	}, nil
+	return result, nil
 }
 
 func testCode(
 	testFiles map[string][]byte,
 	state *flowkit.State,
 	flags flagsTests,
-) (map[string]cdcTests.Results, *runtime.CoverageReport, error) {
-	var coverageReport *runtime.CoverageReport
+) (*result, error) {
 	runner := cdcTests.NewTestRunner()
+
+	var coverageReport *runtime.CoverageReport
 	if flags.Cover {
 		coverageReport = runtime.NewCoverageReport()
 		if flags.CoverCode == contractsCoverCode {
 			coverageReport.WithLocationFilter(
 				func(location common.Location) bool {
 					_, addressLoc := location.(common.AddressLocation)
-					_, stringLoc := location.(common.StringLocation)
-					// We only allow inspection of AddressLocation or StringLocation,
-					// since scripts and transactions cannot be attributed to their
-					// source files anyway.
-					return addressLoc || stringLoc
+					// We only allow inspection of AddressLocation,
+					// since scripts and transactions cannot be
+					// attributed to their source files anyway.
+					return addressLoc
 				},
 			)
 		}
 		runner = runner.WithCoverageReport(coverageReport)
 	}
 
+	var seed int64
+	if flags.Seed > 0 {
+		seed = flags.Seed
+		runner = runner.WithRandomSeed(seed)
+	} else if flags.Random {
+		seed = int64(rand.Intn(150000))
+		runner = runner.WithRandomSeed(seed)
+	}
+
+	contracts := make(map[string]common.Address, 0)
+	for _, contract := range *state.Contracts() {
+		alias := contract.Aliases.ByNetwork("testing")
+		if alias != nil {
+			contracts[contract.Name] = common.Address(alias.Address)
+		}
+	}
+
 	testResults := make(map[string]cdcTests.Results, 0)
 	for scriptPath, code := range testFiles {
 		runner := runner.
 			WithImportResolver(importResolver(scriptPath, state)).
-			WithFileResolver(fileResolver(scriptPath, state))
+			WithFileResolver(fileResolver(scriptPath, state)).
+			WithContracts(contracts)
+
 		results, err := runner.RunTests(string(code))
 		if err != nil {
-			return nil, nil, err
+			return nil, err
 		}
 		testResults[scriptPath] = results
+
 		for _, result := range results {
 			if result.Error != nil {
 				status = 1
 				break
 			}
 		}
 	}
-	return testResults, coverageReport, nil
+
+	return &result{
+		Results:        testResults,
+		CoverageReport: coverageReport,
+		RandomSeed:     seed,
+	}, nil
 }
 
 func importResolver(scriptPath string, state *flowkit.State) cdcTests.ImportResolver {
 	return func(location common.Location) (string, error) {
-		stringLocation, isFileImport := location.(common.StringLocation)
-		if !isFileImport {
-			return "", fmt.Errorf("cannot import from %s", location)
+		contracts := make(map[string]config.Contract, 0)
+		for _, contract := range *state.Contracts() {
+			contracts[contract.Name] = contract
 		}
```

Comment:
Move this out of the returned function (but inside the `importResolver`). i.e: make `contracts` a closure variable, since map creation only needs to be done once per "state". No need to create it everytime the returned import-resolver is invoked

Reply:
Good idea :100: 
Fixed in https://github.com/onflow/flow-cli/pull/1227/commits/efcba45990901b4035c0aa161aced2259191edbb